exchange/bitswap/decision/peer_request_queue.go

```diff
@@ -198,12 +266,20 @@ type activePartner struct {
 
 	// priority queue of tasks belonging to this peer
 	taskQueue pq.PQ
+
+	//  kinda dirty to give each active partner all the ledgers and it's pid
+	//  instead of just its own ledger, but i'm not sure how else to deal with
+	//  the lazily loaded ledgers
```

Comment:
maybe can use one object that has all the ledgers, and we mass a ptr to that into here. the object could be:

```go
type LedgerSet struct {
  ledgers map[peer.ID]*ledger
}

// and then use:
ledgers *LedgerSet
```
or something?